solarized.el

```diff
@@ -126,11 +131,13 @@ COLOR1 and COLOR2 should be color names (e.g. \"white\") or RGB
 triplet strings (e.g. \"#ff12ec\").
 
 Alpha should be a float between 0 and 1."
-  (apply 'color-rgb-to-hex
-         (-zip-with '(lambda (it other)
-                       (+ (* alpha it) (* other (- 1 alpha))))
-                    (color-name-to-rgb color1)
-                    (color-name-to-rgb color2))))
+  (if solarized-termcolors
+      (if (< alpha 0.5) color2 color1)
```

Comment:
I'm not quite certain what this is supposed to do. Some comments might be useful.